pkg/shp/cmd/buildrun/create.go

```diff
@@ -77,15 +88,179 @@ func createCmd() runner.SubCommand {
 		Long:  buildRunCreateLongDesc,
 	}
 
-	// instantiating command-line flags, using an actual BuildRunSpec object to receive the flags
-	// issued on command-line, also marking flags as required
-	buildRunSpecFlags := flags.BuildRunSpecFromFlags(cmd.Flags())
+	createCommand := &CreateCommand{
+		cmd:          cmd,
+		buildRunSpec: flags.BuildRunSpecFromFlags(cmd.Flags()),
+		buildRunOpts: flags.BuildRunOptsFromFlags(cmd.Flags()),
+	}
+
 	if err := cmd.MarkFlagRequired(flags.BuildrefNameFlag); err != nil {
 		panic(err)
 	}
 
-	return &CreateCommand{
-		cmd:          cmd,
-		buildRunSpec: buildRunSpecFlags,
+	return createCommand
+}
+
+// CreateBuildRun creates a BuildRun based on the given name and spec using the
+// provided end-user options/settings, for example printing the pod container
+// logs of the build.
+func CreateBuildRun(
```

Comment:
If I understand it correctly, we've moved the code from `pkg/cmd/build/run.go` to this new place, but why do we have it all convoluted in a single method? Back there we had `onEvent` and better separation of concerns.

Reply:
My problem basically was that the `onEvent` function as a pointer receiver function might have done the trick, however due to the fact that it was "connected" to the `RunCommand` being a pointer receiver function made it unable to be used somewhere else. So it ended up being a function that was only used and could only be used in one exact place.

Moving the code in-line as an ad-hoc function feels more in line with how similar packages deal with those kind of functions, e.g. the Kubernetes `wait` package, or `sort.Slice()`, or similar. Plus, being in-line helps because you do not have to "expose" other fields in order to make them accessible, like the BuildRun name. It being inline gives it access to the correct variables through the scope. This further simplified the code a bit. Also, with that, I was able to improve the output so that it does not print the "Pending" phase multiple times, since the in-line function has access to the helper variable that knows the previous state.

To be honest, I played around with a lot of different options and styles and this only represents the state I was happy with. It would have been probably easier if this was done as part of a pairing session, because just with the PR review one usually lacks a lot of the thought process that was involved.

Reply:
I think we need more attention on this part of the code, more refactoring and more testing as well. So, I'm okay on merging this as it's, and then continue iterating on the next PRs 👍🏼

Reply:
cool / thanks @otaviof 

can you open an issue with any specifics / details you can provide, so we don't lose track ?